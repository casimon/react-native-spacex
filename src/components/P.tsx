import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  StyleProp,
  TextStyle,
  ViewStyle
} from 'react-native';

import colors from '../constants/colors';

const styles = StyleSheet.create({
  p: {
    paddingVertical: 8,
    paddingHorizontal: 6
  },
  pLabel: {
    fontSize: 14,
    color: colors.black,
    fontWeight: 'bold',
    marginBottom: 8,
    textAlign: 'center'
  },
  pText: {
    fontSize: 14,
    color: colors.black,
    textAlign: 'center'
  }
});

interface Props {
  text: string;
  label?: string;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
}

class P extends React.PureComponent<Props> {
  render() {
    return (
      <View style={[styles.p, this.props.style]}>
        {this.props.label && (
          <Text style={styles.pLabel} allowFontScaling={false}>
            {this.props.label}
          </Text>
        )}
        <Text
          style={[styles.pText, this.props.styleText]}
          allowFontScaling={false}>
          {this.props.text}
        </Text>
      </View>
    );
  }
}

export default P;
