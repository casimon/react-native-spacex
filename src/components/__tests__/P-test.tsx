/**
 * @format
 */

import 'react-native';
import React from 'react';
import P from '../P';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders paragraph with text', () => {
  const tree = renderer.create(<P text="Test text prop" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders paragraph with text and optional label prop', () => {
  const tree = renderer
    .create(<P text="Test text" label="Test label" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders paragraph with text and optional style prop', () => {
  const tree = renderer
    .create(<P text="Test text" style={{margin: 2}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders paragraph with text and optional styleText prop', () => {
  const tree = renderer
    .create(<P text="Test text" styleText={{fontSize: 12}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
