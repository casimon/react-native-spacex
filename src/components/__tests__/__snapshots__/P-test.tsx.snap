// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders paragraph with text 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 14,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test text prop
  </Text>
</View>
`;

exports[`renders paragraph with text and optional label prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginBottom": 8,
        "textAlign": "center",
      }
    }
  >
    Test label
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 14,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test text
  </Text>
</View>
`;

exports[`renders paragraph with text and optional style prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      },
      Object {
        "margin": 2,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 14,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Test text
  </Text>
</View>
`;

exports[`renders paragraph with text and optional styleText prop 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 14,
          "textAlign": "center",
        },
        Object {
          "fontSize": 12,
        },
      ]
    }
  >
    Test text
  </Text>
</View>
`;
